src/react-freshchat.js

```diff
@@ -136,6 +136,15 @@ class FreshChat extends React.Component {
   componentWillUnmount() {
     widget().close()
   }
+
+  componentWillReceiveProps(nextProps) {
+    let { token, ...moreProps } = nextProps;
+    this.init({
```

Comment:
Aren't class methods not automatically bound to `this`? Does this actually work?